plugins/woocommerce/tests/e2e-pw/tests/shopper/shop-title-after-deletion.spec.js

```diff
@@ -8,6 +8,8 @@ test.describe(
 		test.use( { storageState: process.env.ADMINSTATE } );
 		test.beforeEach( async ( { page } ) => {
 			await page.goto( 'wp-admin/edit.php?post_type=page' );
+			await page.locator( 'input[name="s"]' ).fill( 'Shop' );
+			await page.locator( 'input[value="Search Pages"]' ).click();
```

Comment:
This is to first search for Shop and then click on it. Sometimes, there are pages left, and the Shop becomes visible on the second page.

Reply:
@adimoldovan  The pages? Yes, they should be all auto-deleted, I have no idea why they were left. When I executed all the shopper tests from the local, no pages where left. It seems that sometimes flaky/failing test (shopper or merchant) leaves behind some created pages such as checkout-blocks-12382 etc.

With this workaround to search for the shop page first, we will ensure the shop page will be found before clicking it.

Reply:
Ah, that's worth updating, as it looks pretty unnecessary.

I'm trying locally to make it work using `wpApi` and I know there is fixtures one, however, even with this current approach of using `beforeAll` & `afterAll` [I don't know why it won't update the page status](https://d.pr/i/SNMvSj).

When I added simple code to check if my post api works, it says: Error: Failed to update shop page status. Status: 401

The same is even for wp-end, not just external envs. It might be some problem to auth to post... Do you have any advice on how to proceed? @adimoldovan

Reply:
Omg, why I wanted to make the status `draft`, actually I wanted to put the page in the trash. My bad, I will try it that way.

Reply:
@veljkho 
- you get a 401 error, which means you probably set the api client wrong. Why are you not using the `wpApi` fixture instead? It's there for exactly this.
- I don't think you can trash a page by setting it's status. You need to delete it without the `force` argument set to `true`.

Reply:
Here's the code with the fixture that works fine locally. Try using this instead of before/after hooks. It doesn't work with the pressable site because it cannot find the page with `slug=shop`. Probably because it was deleted in previous test runs and remained so? You'll need to figure that part out.

```
const { test: baseTest, expect } = require( '../../fixtures/fixtures' );

// test case for bug https://github.com/woocommerce/woocommerce/pull/46429
const test = baseTest.extend( {
	storageState: process.env.ADMINSTATE,
	page: async ( { page, wpApi }, use ) => {
		const response = await wpApi.get( `/wp-json/wp/v2/pages?slug=shop`, {
			data: {
				_fields: [ 'id' ],
			},
		} );

		const pages = await response.json();
		console.log( pages );
		const pageId = pages[ 0 ].id;

		await wpApi.delete( `/wp-json/wp/v2/pages/${ pageId }`, {
			data: {
				force: false,
			},
		} );

		await use( page );

		await wpApi.post( `/wp-json/wp/v2/pages/${ pageId }`, {
			data: {
				status: 'publish',
			},
		} );
	},
} );

test.describe(
	'Check the title of the shop page after the page has been deleted',
	{ tag: [ '@payments', '@services', '@could-be-unit-test' ] },
	() => {
// rest of the code
```

Reply:
Thanks! This is working for me as well. I remember trying to involve fixtures and even `pages?slug=shop` but not to delete it without `force: false`.

I just added a new commit improving the test using `wpApi` fixture and it works like a charm.

Reply:
The shop page on Pressable is in draft status. I remember fixing that already, and it should not occur again.

Reply:
Actually, fixed it in this PR, now with the new code using fixtures:
```
await wpApi.post( `/wp-json/wp/v2/pages/${ pageId }`, {
	data: {
		status: 'publish',
	},
} );
```

Reply:
> The shop page on Pressable is in draft status.

Do we know how/when this happened?

Reply:
The fix for that is in this PR. Someone executed tests without the fix from this PR of course.

Reply:
That might be me :)

Reply:
The original test `shop-title-after-deletion.spec.js` didn't clean up after moving the shop page from trash (restoring). Then, I added it via UI to make it published. However with the above `wpApi`, we have it as fixed again.